sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -288,6 +290,27 @@ static constexpr bool is_function_ptr_or_ref_v =
 #endif // __INVOKE_SIMD_USE_STD_IS_FUNCTION_WA
     ;
 
+/// The variables typed as pointer to a function become lvalue-reference
+/// when passed to invoke_simd as universal pointer. That creates an
+/// additional indirection, which is resolved automatically by the compiler
+/// for the caller side of __builtin_invoke_simd, but which must be resolved
+/// manually during the creation of invoke-simd-helper.
+/// The class remove_ref_if_ref_to_pointer_callable helps to removes that
+/// unwanted indirection for variable-pointers to functions passed to
+/// invoke_simd.
```

Comment:
I suggest to move this comment to the "call site", and leave here just a short description like "remove reference from 'reference to function pointer' type".